.github/workflows/build.yml

```diff
@@ -31,58 +31,50 @@ jobs:
       matrix:
         build_type: [Release,Debug]
         name:
-          - linux gnu-10
-          - linux clang-12
-          - linux nvhpc-21.9
-          - linux intel-classic
+          - linux gnu-13
+          - linux clang-18
+          - linux nvhpc-25.1
+          - linux intel-llvm
           - macos
 
         include:
 
-          - name: linux gnu-10
-            os: ubuntu-20.04
-            compiler: gnu-10
-            compiler_cc: gcc-10
-            compiler_cxx: g++-10
-            compiler_fc: gfortran-10
-            ctest_options: -E memory
-            caching: true
-
-          - name: linux clang-12
-            os: ubuntu-20.04
-            compiler: clang-12
-            compiler_cc: clang-12
-            compiler_cxx: clang++-12
-            compiler_fc: gfortran-10
+          - name: linux gnu-13
+            os: ubuntu-24.04
+            compiler: gnu-13
+            compiler_cc: gcc-13
+            compiler_cxx: g++-13
+            compiler_fc: gfortran-13
             ctest_options: -E memory
             caching: true
 
-          - name: linux clang-12
-            build_type: Release
-            os: ubuntu-20.04
-            compiler: clang-12
-            compiler_cc: clang-12
-            compiler_cxx: clang++-12
-            compiler_fc: gfortran-10
+          - name: linux clang-18
+            os: ubuntu-24.04
+            compiler: clang-18
+            compiler_cc: clang-18
+            compiler_cxx: clang++-18
+            compiler_fc: gfortran-14
             ctest_options: -E memory
             caching: true
 
-          - name: linux nvhpc-21.9
-            os: ubuntu-20.04
-            compiler: nvhpc-21.9
+          - name: linux nvhpc-25.1
+            os: ubuntu-24.04
+            compiler: nvhpc-25.1
             compiler_cc: nvc
             compiler_cxx: nvc++
             compiler_fc: nvfortran
             cmake_options: -DCMAKE_CXX_FLAGS=--diag_suppress177
             ctest_options: -E memory
             caching: false
 
-          - name : linux intel-classic
-            os: ubuntu-20.04
-            compiler: intel-classic
-            compiler_cc: icc
-            compiler_cxx: icpc
-            compiler_fc: ifort
+          - name : linux intel-llvm
+            os: ubuntu-24.04
+            compiler: intel-llvm
+            compiler_cc: icx
+            compiler_cxx: icpx
+            compiler_fc: ifx
+             # TODO: this tests is broken with intel-llvm (segfault)
+            ctest_options: -E ectrans_test_install
```

Comment:
ectrans_test_install fails??? That is very concerning...

Reply:
Yeah, I'll reenable it so you can see the error.

Reply:
See https://github.com/ecmwf-ifs/ectrans/actions/runs/13815938473/job/38649010125?pr=224#step:12:4336. Tricky to debug non-interactively.

Reply:
I had a go on Atos with the same version of Intel (2023.2.0) but the test passed. I think we'll need to debug this through the action itself.